NEWS.md

```diff
@@ -2,8 +2,22 @@
 
 ## polars (development version)
 
+This is an update that corresponds to Python Polars 1.36.0, which includes significant internal changes.
```

Comment:
Not completely related to this PR but given the amount of big changes in 1.36.0, I guess there will be a 1.36.1 shortly after and we should target that for our next release.